### PR TITLE
Feature/jjdd8fpo 11

### PR DESCRIPTION
**Pull request by lukaszwolek**

### CODE REVIEW

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: src/main/java/com/infoshareacademy/BookFinder.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import java.util.stream.Collectors;
+
+
+public class BookFinder {
+
+    private static final Logger stdout = LoggerFactory.getLogger("CONSOLE_OUT");
+
+    final static List<Book> BOOKS = BookRepository.getInstance().getBooks();
+
+    public void bookFinderSetup(int param, int hasAudio) {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookFinder.java]: i znowu rzeczownik ;-)

File: src/main/java/com/infoshareacademy/Menu.java [hunk 2]
```diff
@@ -40,7 +40,7 @@ public void showMenu(int position) {
 
         while (position != EXIT_POSITION) {
             ScreenCleaner.clearScreen();
-            Header.headerPrinter();
+            //Header.headerPrinter();
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/Menu.java]: nie wrzucamy zakomentowanego kodu do repo :-)

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

File: src/main/java/com/infoshareacademy/BookFinder.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                titleslist = findTitleByName(getLetters());
+                title = verifyFindingAuthor(titleslist);
+            }
+            printFilteredBooks(param, hasAudio, "", title);
+        }
+        if (param == 3) {
+            stdout.info("\nSzukanie książek po autorze i tytule podając najpierw ciąg znaków który zawiera się \n w tytule , następnie ciąg znaków zawierający się w imieniu lub nazwisku autora \n");
+            List<String> authorslist;
+            while (author.isEmpty()) {
+                authorslist = findAuthorByName(getLetters());
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookFinder.java]: camel case w nazwie

File: src/main/java/com/infoshareacademy/BookFinder.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        if (param == 2) {
+            stdout.info("\nSzukanie ksiązek po tytule podając ciag znaków który zawiera się w tytule \n ");
+            List<String> titleslist;
+            while (title.isEmpty()) {
+                titleslist = findTitleByName(getLetters());
+                title = verifyFindingAuthor(titleslist);
+            }
+            printFilteredBooks(param, hasAudio, "", title);
+        }
+        if (param == 3) {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookFinder.java]: te liczby w ifach powinny być stałymi opisującymi co dokładnie oznaczają ponieważ nic nie można z tego wywnioskować

File: src/main/java/com/infoshareacademy/BookFinder.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+        }
+        return;
+    }
+
+    private void printFilteredBooks(int searchingMethod, int hasAudio, String author, String title) {
+
+        List<Book> filteredBooks = BOOKS;
+
+        if (hasAudio == 1) {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookFinder.java]: I znowu te cyferki :) przenieśmy je do stałych bo same w sobie nic nie znaczą

File: src/main/java/com/infoshareacademy/BookFinder.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        } else if (searchingMethod == 3) {
+            filteredBooks = filteredBooks.stream()
+                    .filter(Objects::nonNull)
+                    .filter(b -> b.getTitle().equals(title))
+                    .filter(b -> b.getAuthor().equals(author))
+                    .collect(Collectors.toList());
+
+            filteredBooks.forEach(x -> stdout.info(String.valueOf(x)));
+        }
+        if (filteredBooks.isEmpty()) {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookFinder.java]: Dlaczego ten if nie jest jako pierwszy w tej metodzie? mógłby wówczas zawierać instrukcję return bez konieczności wykonywania wszystkich pozostałych ifów

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by DariuszCzajkowski** [comment]
z tymi konfliktami to nie przejdzie

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Pull request closed without merging**